mne/tests/test_surface.py

```diff
@@ -405,3 +411,37 @@ def test_warp_montage_volume():
     with pytest.warns(RuntimeWarning, match='not assigned'):
         warp_montage_volume(doubled_montage, CT, reg_affine,
                             None, 'sample', subjects_dir_from=subjects_dir)
+
+
+@testing.requires_testing_data
+@pytest.mark.parametrize('ret_nn', (False, True))
+@pytest.mark.parametrize('method', ('accurate', 'nearest'))
+def test_project_onto_surface(method, ret_nn):
+    """Test _project_onto_surface (gh )."""
+    trans = read_trans(
+        data_path / 'MEG' / 'sample' / 'sample_audvis_trunc-trans.fif')
+    surf = read_bem_surfaces(
+        data_path / 'subjects' / 'sample' / 'bem' / 'sample-head.fif')[0]
+    head_surf = transform_surface_to(
+        surf=surf, dest='mri', trans=_get_trans(trans, 'head', 'mri'),
+        copy=True)
+    # set up the raw
+    montage = make_standard_montage('GSN-HydroCel-129')
+    info = create_info(montage.ch_names, 100, ch_types='eeg')
+    raw = RawArray(np.zeros((len(montage.ch_names), 100)), info)
+    raw.set_montage(montage)
+    # get MRI fiducials
+    mri_fiducials = get_mni_fiducials('sample', data_path / 'subjects')
+    mri_fids, frame = _get_fid_coords(mri_fiducials)
+    mri_fid_loc = np.array(list(mri_fids.values()))
+    # get montage fiducials
+    cap_fids, frame = _get_fid_coords(raw.info['dig'])
```

Comment:
@christian-oreilly just FYI, `_get_fid_coords` is a useful private func for testing, to replace the list comprehensions you were using

Reply:
Thanks for the tip. Nice thinking for generalizing the test (i.e., `@pytest.mark.parametrize('ret_nn', (False, True))`, 
`@pytest.mark.parametrize('method', ('accurate', 'nearest')))`.